### ISSUE TITLE
Better Calico compatibility in a multi-cni configuration

### ISSUE THREAD

**Issue by redteamer-io-dev** [projectcalico/calico#5199]
## Expected Behaviour
Calico to be extra ;) useful as delegate in a multi cni configuration. E.G only used for pod and service networking in a Kubernetes context.

## Current Behavior
Calico CNI is hard coded to be the default gateway https://github.com/projectcalico/cni-plugin/blob/bcd2d3f825002c218bfaa872206f9bf42753c7e3/utils/network.go#L100 
It would be great if there was a boolean option exposed in Calico's CNI [NetConf](https://github.com/projectcalico/cni-plugin/blob/28a93c7f17bae5b3ea6347c509b52066f069241b/types/types.go#L59). that allowed users to specify if Calico should be the default gateway

## Possible Solution
Add an additional field (isDefaultGateway bool) with a default value of true in the NetConf type  https://github.com/projectcalico/cni-plugin/blob/28a93c7f17bae5b3ea6347c509b52066f069241b/types/types.go#L59 

Wrap a `if conf.isDefaultGateway == true {}` around the following line https://github.com/projectcalico/cni-plugin/blob/bcd2d3f825002c218bfaa872206f9bf42753c7e3/utils/network.go#L100

## Context
I require multiple interfaces exposed to Kubernetes Pods where each pods primary interface is used for Pod and Service networking and a second interface utilised for public internet connectivity. An example use-case for this configuration is independent metering of East/West and North/South traffic that's only possible when multiple interfaces are attached to a pod.

**Comment by maxpain** [comment]
Any updates?

**Comment by toelke** [comment]
> This has been working for me in production since I contributed to this ticket, though i'd still prefer to not need a custom CNI that has to undo work done by calico.

Do you happen to have the code for that custom CNI still lying around?

**Comment by remram44** [comment]
Definitely still interested in this! Multi-CNI is pretty common now

**Comment by groundsada** [comment]
Yes, please revive this issue, this is highly relevant for our cluster.
